scripts/get_orphaned_functions.py

```diff
@@ -0,0 +1,246 @@
+    """
+
+    # Regex pattern to detect usage of the function
+    function_pattern = re.compile(r"\b" + re.escape(function_name) + r"\b\s*\(")
+
+    # Traverse all files in the given module R directory
+    for Rfile in os.listdir(MODULE_DIRECTORY_PATH):
+        # Check if the file is an R file
+        if Rfile.endswith(".R"):
+            file_path = os.path.join(MODULE_DIRECTORY_PATH, Rfile)
+            with open(file_path, "r") as file:
+                file_content = file.read()
+
+                # Checks if the function is utilized in the file by searching for the function name
+                if function_pattern.search(file_content):
+                    # print(f'{color_codes[2]}Function "{function_name}" is utilized in file: {file_path}{ENDC}')
+                    return True
+
+    # If no file uses the function
+    # print(f'{color_codes[1]}Function "{function_name}" is not utilized in module: {MODULE_DIRECTORY_PATH}, {ENDC}')
+    return False
+
+
+def is_function_utilized_externally(function_name, module_name):
+    """
+    Check if the function from a specific module is utilized in any other module.
+
+    Args:
+    function_name (str): The name of the function to check.
+    module_name (str): The name of the module where the function is originally defined.
+
+    Returns:
+    bool: True if the function is utilized externally, False otherwise.
+    """
+    func_location_module = os.path.join(main_dir, "modules", module_name)
+    search_directories = [
+        "modules",
+        "models",
+        "scripts",
+        "base",
+        "tests",
+        "base",
+    ]
+
+    # print(color_codes[2] + "Function name:",function_name," \t Function's Location modules :",func_location_module,ENDC,)
+
+    for directory in search_directories:
+        complete_dir_path = os.path.join(main_dir, directory)
+        for MODULE_DIRECTORY_PATH, dirs, R_files in os.walk(complete_dir_path):
+            if MODULE_DIRECTORY_PATH.__contains__(func_location_module) or MODULE_DIRECTORY_PATH.__contains__("/man"):
+                # print(f"{color_codes[13]} Skipping the module directory {MODULE_DIRECTORY_PATH}{ENDC}")
+                continue
+            else:
+                if is_function_utilized_within_module(function_name, MODULE_DIRECTORY_PATH):
+                    # print(f'{color_codes[12]} Function "{function_name}" is utilized EXTERNALLY in module: {MODULE_DIRECTORY_PATH}{ENDC}')
+                    return True
+    return False
+
+
+def print_in_batches(list_to_print, label):
+    # print(label)
+    color_index = 0  # Start with the first color
+    for i in range(0, len(list_to_print), 5):
+        batch = list_to_print[i : i + 5]
+        # Get the current color
+        color = color_codes[color_index % len(color_codes)]
+        # Print the batch in the current color
+        print(color + ", ".join(map(str, batch)) + ENDC)
+        # Move to the next color for the next batch
+        color_index += 1
```

Comment:
Maybe We can remove this Function.
I created this function for better printing in the Terminal so that readability issues are removed.
However, we may still find their usage if anyone tried debugging this code.

So, shall I remove this or Keep this for future use?